trio/_core/_run.py

```diff
@@ -1243,86 +1242,8 @@ def spawn_impl(self, async_fn, args, nursery, name, *, system_task=False):
         if nursery is None:
             assert self.init_task is None
 
-        ######
-        # Call the function and get the coroutine object, while giving helpful
```

Comment:
This is a better description than the "Check if async_fn is callable coroutine" that you added below, so I'd suggest you keep it. (The function is not a "callable coroutine" -- it's a callable that returns a coroutine object, but that's just another way of saying "async function" which is Trio's preferred term. Coroutine objects themselves are not callable.)